gslib/commands/rm.py

```diff
@@ -393,10 +393,15 @@ def RemoveFunc(self, name_expansion_result, thread_state=None):
 
     exp_src_url = name_expansion_result.expanded_storage_url
     self.logger.info('Removing %s...', exp_src_url)
-    gsutil_api.DeleteObject(exp_src_url.bucket_name,
-                            exp_src_url.object_name,
-                            preconditions=self.preconditions,
-                            generation=exp_src_url.generation,
-                            provider=exp_src_url.scheme)
+    try:
+      gsutil_api.DeleteObject(exp_src_url.bucket_name,
+                              exp_src_url.object_name,
+                              preconditions=self.preconditions,
+                              generation=exp_src_url.generation,
+                              provider=exp_src_url.scheme)
+    except NotFoundException as e:
+      # Retries during recursive object deletes can cause harmless 404s.
+      self.logger.info('Cannot find %s', exp_src_url)
+      DecrementFailureCount()
```

Comment:
Why do we have call this? Is DeleteObject call going to increment the failure count?

Reply:
`command.py` increments failure count in `PerformTask`. This score keeping would be off by one when the 404 is harmless. I'll add a slightly more verbose comment to explain why this happens.